packages/data-table-server/src/dataTable/AbstractDataTable.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { TupaiaApiClient } from '@tupaia/api-client';
+import { yup } from '@tupaia/utils';
+
+export abstract class AbstractDataTable<
+  ParamsSchema extends yup.AnySchema = yup.AnySchema,
+  ConfigSchema extends yup.AnySchema = yup.AnySchema,
+  RecordSchema = unknown
+> {
+  protected readonly paramsSchema: ParamsSchema;
+  protected readonly configSchema: ConfigSchema;
+  protected readonly apiClient: TupaiaApiClient;
+  protected readonly config: yup.InferType<ConfigSchema>;
+
+  protected constructor(
+    paramsSchema: ParamsSchema,
+    configSchema: ConfigSchema,
+    apiClient: TupaiaApiClient,
```

Comment:
Passing through the `apiClient` is required for the `AnalyticsDataTable`, but it already feels like a bit of dependency pollution... I'll hold off engineering anything to handle it gracefully until we've got a few more data-table types out and we can see what's common?